frb_codegen/src/ir/ty.rs

```diff
@@ -26,6 +29,25 @@ impl IrType {
         self.visit_children_types(f, ir_file);
     }
 
+    pub fn distinct_types(&self, ir_file: &IrFile) -> Vec<IrType> {
+        let mut seen_idents = HashSet::new();
+        let mut ans = Vec::new();
```

Comment:
uber-nit: what about using a ordered hash set (uber nit so it is ok not to do so)